lessons/advanced/typespec.md

```diff
@@ -0,0 +1,197 @@
+---
+layout: page
+title: Specifications and types
+category: advanced
```

Comment:
@doomspork, `typespec` in `advanced` and Dialyzer/Dialyxir in `specifics`, because typespec is an element of language, but D&D are tools.